src/user/bans.js

```diff
@@ -63,33 +61,47 @@ module.exports = function (User) {
 		], callback);
 	};
 
-	User.isBanned = function (uid, callback) {
+	User.getBannedAndExpired = function (uid, callback) {
 		if (parseInt(uid, 10) <= 0) {
 			return setImmediate(callback, null, false);
 		}
-		async.waterfall([
-			async.apply(User.getUserFields, uid, ['banned', 'banned:expire']),
-			function (userData, next) {
-				var banned = userData && userData.banned;
-				if (!banned) {
-					return next(null, banned);
-				}
+		User.getUserFields(uid, ['banned', 'banned:expire'], function (err, userData) {
+			if (err) {
+				return callback(err);
+			}
+			callback(null, User.calcBanExpiredFromUserData(userData));
+		});
+	};
 
-				// If they are banned, see if the ban has expired
-				var stillBanned = !userData['banned:expire'] || Date.now() < userData['banned:expire'];
+	User.calcBanExpiredFromUserData = function (userData) {
```

Comment:
I feel like `getBanInfo` is a little more succinct?

Reply:
true, but I needed a synchronous function here since in both places where I needed to figure out if the ban expired, I already had the data for it.. , also didn't want to confuse it with the existing getBanInfo or getLatestBanInfo.. 

let me try to clean it up a bit

Reply:
wait, are you suggesting that I actually use the getLatetBanInfo to do the auto-unban?